python/ray/_private/storage.py

```diff
@@ -368,6 +369,9 @@ def _init_filesystem(create_valid_file: bool = False, check_valid_file: bool = T
         fs_creator = _load_class(parsed_uri.netloc)
         _filesystem, _storage_prefix = fs_creator(parsed_uri.path)
     else:
+        # Arrow's S3FileSystem doesn't allow creating buckets by default, so we add a
+        # query arg enabling bucket creation if an S3 URI is provided.
+        _storage_uri = _add_creatable_buckets_param_if_s3_uri(_storage_uri)
```

Comment:
Not sure the exact semantics of ray.init(storage=uri), but is this desired to always create bucket?

Reply:
Those were the old semantics, and we have tests relying on that behavior, so this is just preserving the existing semantics of creating a bucket if necessary.